### PR TITLE
Fix double purify bug

### PR DESCRIPTION
**Pull request by KiwiTrinsic**
Fixing a bug that could prevent regaining starting traits when purifying a mutation that overwrote one. Also fixed an unrelated typo in mutations.json that was discovered in the process of hunting the bug down.

### CODE REVIEW

File: src/mutation.cpp [hunk 1]
```diff
@@ -1271,6 +1271,11 @@ void player::remove_mutation( const std::string &mut )
         }
     }
 
+    // make sure we don't toggle a mutation or trait twice, or it will cancel itself out.
+    if(replacing == replacing2) {
+        replacing2 = "";
+    }
+
     // This should revert back to a removed base trait rather than simply removing the mutation
```
Comment (DavidKeaton) [inline src/mutation.cpp]: isn't this what you were fixing? is this behaviour intentional.

**Comment by KiwiTrinsic** [comment]
The problem is that if the same trait is in replacing and replacing2, it will get toggled twice... which leaves it toggled off.

**Comment by kevingranade** [comment]
I made a character with the smelly trait, gave him flowers, which cancels smelly, then removed flowers, but smelly didn't come back.  I'm not sure if I'm doing something wrong or if this didn't fix the issue or if it's another issue.

**Comment by Coolthulhu** [comment]
Doesn't fix all the problems, but is still better than what we had in master.

**Comment by KiwiTrinsic** [comment]
@kevingranade That looks like a another issue. The fix I put in only effects reverting to starting traits. The test I was using with it was creating a character with the Pretty trait, mutating to Ugly, then drinking a purifier to revert.

**Pull request merged**